src/components/form/form.component.js

```diff
@@ -0,0 +1,507 @@
+    this._window.removeEventListener('scroll', this.checkStickyFooter);
+  }
+
+  checkStickyFooter = () => {
+    if (!this._form) { return; }
+
+    let offsetTop = 0,
+        element = this._form;
+
+    while (element) {
+      offsetTop += element.offsetTop;
+      element = element.offsetParent;
+    }
+
+    const formHeight = (offsetTop + this._form.offsetHeight) - this._window.pageYOffset;
+
+    if (!this.state.stickyFooter && formHeight > this._window.innerHeight) {
+      this.setState({ stickyFooter: true });
+    } else if (this.state.stickyFooter && formHeight < this._window.innerHeight) {
+      this.setState({ stickyFooter: false });
+    }
+  }
+
+  addUnsavedWarningListener = () => {
+    this._window.addEventListener('beforeunload', this.checkIsFormDirty);
+  }
+
+  removeUnsavedWarningListener = () => {
+    this._window.removeEventListener('beforeunload', this.checkIsFormDirty);
+  }
+
+  // This must return undefined for IE and Safari if we don't want a warning
+  /* eslint-disable consistent-return */
+  checkIsFormDirty = (ev) => {
+    if (this.state.isDirty) {
+      // Confirmation message is usually overridden by browsers with a similar message
+      const confirmationMessage = I18n.t('form.save_prompt',
+        { defaultValue: 'Do you want to leave this page? Changes that you made may not be saved.' });
+      ev.returnValue = confirmationMessage; // Gecko + IE
+      return confirmationMessage; // Gecko + Webkit, Safari, Chrome etc.
+    }
+  }
+  /* eslint-enable consistent-return */
+
+  /**
+   * stores the document - allows us to override it different contexts, such as
+   * when running tests.
+   */
+  _document = Browser.getDocument();
+
+  /**
+   * stores the window - allows us to override it different contexts, such as
+   * when running tests.
+   */
+  _window = Browser.getWindow();
+
+  /**  Handles submit and runs validation. */
+  handleOnSubmit = async (ev) => {
+    ev.preventDefault();
+
+    if (this.props.autoDisable) {
+      this.setState({ submitted: true });
+    }
+
+    if (this.props.beforeFormValidation) {
+      this.props.beforeFormValidation(ev);
+    }
+
+    const valid = await this.props.validate();
+
+    if (this.props.afterFormValidation) {
+      this.props.afterFormValidation(ev, valid, this.enableForm);
+    }
+
+    if (valid) {
+      this.resetIsDirty();
+      this.triggerSubmit(ev, valid);
+    } else {
+      this.setState({ submitted: false });
+    }
+  }
+
+  triggerSubmit(ev, valid) {
+    if (this.props.onSubmit) {
+      this.props.onSubmit(ev, valid, this.enableForm);
+    } else {
+      this._form.submit();
+    }
+  }
+
+  /** enables a form which has been disabled after being submitted. */
+  enableForm = () => {
+    this.setState({ submitted: false });
+  }
+
+  /**
+   * Serializes the inputs in the form ready for submission via AJAX
+   * https://www.npmjs.com/package/form-serialize
+   */
+  serialize = (opts) => {
```

Comment:
Yeah I was thinking the same thing, I'm going to leave it in the old Form as it may break some legacy stuff removing it. But I'll remove it form the experimental implementation

Reply:
OK, sounds good, thanks :white_check_mark: